tck/src/main/java/com/hedera/hashgraph/tck/methods/AbstractJSONRPC2Service.java

```diff
@@ -0,0 +1,110 @@
+import com.hedera.hashgraph.tck.exception.InvalidJSONRPC2ParamsException;
+import com.hedera.hashgraph.tck.methods.JSONRPC2Error.ErrorData;
+import com.thetransactioncompany.jsonrpc2.JSONRPC2Error;
+import com.thetransactioncompany.jsonrpc2.JSONRPC2Request;
+import com.thetransactioncompany.jsonrpc2.JSONRPC2Response;
+import com.thetransactioncompany.jsonrpc2.server.MessageContext;
+import com.thetransactioncompany.jsonrpc2.server.RequestHandler;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Implements RequestHandler and overrides some of the Dispatcher logic,
+ * enhancing the usability of the process method.
+ *
+ * Keeps track of the {@link JSONRPC2Method} annotated methods in a map,
+ * converts the JSON-RPC params to the needed args in the called method
+ * and invokes it.
+ */
+public abstract class AbstractJSONRPC2Service implements RequestHandler {
+
+    // this is shared state to all requests so there could be race conditions
+    // although the tck driver would not call these methods in such way
+    private final Map<String, Method> methodMap;
+
+    public AbstractJSONRPC2Service() {
+        methodMap = new HashMap<>();
+        registerMethods();
+    }
+
+    /**
+     * Register JSONRPCMethods
+     */
+    private void registerMethods() {
+        Method[] methods = getClass().getDeclaredMethods();
+        for (Method method : methods) {
+            if (method.isAnnotationPresent(JSONRPC2Method.class)) {
+                JSONRPC2Method annotation = method.getAnnotation(JSONRPC2Method.class);
+                methodMap.put(annotation.value(), method);
+            }
+        }
+    }
+
+    @Override
+    public String[] handledRequests() {
+        return methodMap.keySet().toArray(new String[0]);
+    }
+
+    /**
+     *
+     * @param req JSON-RPC request
+     * @param messageContext
+     * @return JSONRPC2Response - result returned from the JSONRPCMethod or if error is thrown - JSONRPC2Error
+     */
+    @Override
+    public JSONRPC2Response process(final JSONRPC2Request req, final MessageContext messageContext) {
+        try {
+            Method method = methodMap.get(req.getMethod());
+            if (method != null) {
+                Object[] args = getArguments(method, req.getNamedParams());
+                Object result = method.invoke(this, args);
+                return new JSONRPC2Response(result, req.getID());
+            } else {
+                return new JSONRPC2Response(JSONRPC2Error.METHOD_NOT_FOUND, req.getID());
+            }
+        } catch (InvalidJSONRPC2ParamsException e) {
+            return new JSONRPC2Response(JSONRPC2Error.INVALID_PARAMS, req.getID());
+        } catch (InvocationTargetException e) {
+            // the target is HederaException
+            var hederaException = e.getTargetException();
```

Comment:
The exception handling here will be enhanced to handle precheck and execution errors. This will be done in future PR with account methods implementation.